resources/js/core/twitch-player.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { fadeOut } from 'utils/fade';
+import TurbolinksReload from './turbolinks-reload';
+
+declare global {
+  interface TwitchPlayerConstructor {
+    new (id: string, options: Record<string, unknown>): TwitchPlayerClass;
+    PLAY: string;
+  }
+
+  interface TwitchPlayerClass {
+    addEventListener(action: string, callback: () => void): void;
+  }
+
+  interface Window {
+    Twitch?: {
+      Embed: unknown; // don't care.
+      Player: TwitchPlayerConstructor;
+    };
+  }
+}
```

Comment:
use class declaration instead?

```typescript
declare class TwitchEmbedPlayer {
  static PLAY: string;

  constructor (id: string, options: Record<string, unknown>);
  addEventListener(action: string, callback: () => void): void;
}

declare global {
  interface Window {
    Twitch?: {
      Embed: unknown; // don't care.
      Player: typeof TwitchEmbedPlayer;
    };
  }
}
```

Reply:
declaring it as class sets off eslint `max-classes-per-file` unless ignored and moving the declaration to a separate `.d.ts` adds it to the global namespace instead of limiting it to the file, so there's this roundabout workaround :|

Reply:
Do it like `shopify-buy.d.ts` then...?